Library/Homebrew/cmd/update-bash.sh

```diff
@@ -0,0 +1,256 @@
+#!/bin/bash
+
+if [ -z "$HOMEBREW_BREW_FILE" ]
+then
+  echo "Error: $(basename "$0") must be called from brew!" >&2
+  exit 1
+fi
+
+brew() {
+  "$HOMEBREW_BREW_FILE" "$@"
+}
+
+which_git() {
+  local which_git
+  which_git="$(which git 2>/dev/null)"
+  if [ -n "$which_git" ] && [ "/usr/bin/git" = "$which_git" ]
+  then
+    local active_developer_dir
+    active_developer_dir="$('/usr/bin/xcode-select' -print-path 2>/dev/null)"
+    if [ -n "$active_developer_dir" ] && [ -x "$active_developer_dir/usr/bin/git" ]
+    then
+      which_git="$active_developer_dir/usr/bin/git"
+    else
+      which_git=""
+    fi
+  fi
+  echo $which_git
+}
+
+git_init_if_necessary() {
+  if [ -d ".git" ]
+  then
+    git init -q
+    git config core.autocrlf false
+    git config remote.origin.url https://github.com/Homebrew/homebrew.git
+    git config remote.origin.fetch +refs/heads/*:refs/remotes/origin/*
+  fi
+
+  if git remote show origin -n | grep -q "mxcl/homebrew"
+  then
+    git remote set-url origin https://github.com/Homebrew/homebrew.git
+    git remote set-url --delete origin ".*mxcl\/homebrew.*"
+  fi
+}
+
+repo_var() {
+  echo "$@" |
+    sed -e "s|$HOMEBREW_PREFIX||g" \
+        -e 's|Library/Taps/||g' \
+        -e 's|[^a-z0-9]|_|g' |
+    tr "[:lower:]" "[:upper:]"
+}
+
+upstream_branch() {
+  local upstream_branch
+  upstream_branch="$(git symbolic-ref refs/remotes/origin/HEAD 2>/dev/null |
+                     sed -e 's|refs/remotes/origin/||' )"
+  [ -z "$upstream_branch" ] && upstream_branch="master"
+  echo "$upstream_branch"
+}
+
+read_current_revision() {
+  git rev-parse -q --verify HEAD
+}
+
+pop_stash() {
+  [ -z "$STASHED" ] && return
+  git stash pop $QUIET_ARGS
+  if [ -n "$HOMEBREW_VERBOSE" ]
+  then
+    echo "Restoring your stashed changes to $DIR:"
+    git status --short --untracked-files
+  fi
+  unset STASHED
+}
+
+pop_stash_message() {
+  [ -z "$STASHED" ] && return
+  echo "To restore the stashed changes to $DIR run:"
+  echo "  'cd $DIR && git stash pop'"
+  unset STASHED
+}
+
+reset_on_interrupt() {
+  [ -z "$INITIAL_BRANCH" ] || git checkout "$INITIAL_BRANCH"
+  git reset --hard "$INITIAL_REVISION" $QUIET_ARGS
+  if [ -n "$INITIAL_BRANCH" ]
+  then
+    pop_stash
+  else
+    pop_stash_message
+  fi
+}
+
+pull() {
+  cd "$DIR" || return
+  TAP_VAR=$(repo_var "$DIR")
+  unset STASHED
+
+  # The upstream repository's default branch may not be master;
+  # check refs/remotes/origin/HEAD to see what the default
+  # origin branch name is, and use that. If not set, fall back to "master".
+  INITIAL_BRANCH="$(git symbolic-ref --short HEAD 2>/dev/null)"
+  UPSTREAM_BRANCH="$(upstream_branch)"
+  export HOMEBREW_UPDATE_AFTER"$TAP_VAR"="$(git rev-parse "$UPSTREAM_BRANCH")"
```

Comment:
Why is this set so early in `pull`? Shouldn't this be postponed to the point where `$CURRENT_REVISION` is set for the last time? (Haven't tested this yet, but it seems it will contain the revision from before the merge, so the report later on will be always empty?!)